### PR TITLE
Upgraded: Spring boot upgrade to 2.7.7

### PR DESCRIPTION
**Pull request by vilasshelke-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-6249
https://tools.hmcts.net/jira/browse/RDCC-6537

### Change description ###

1. Spring boot upgrade to 2.7.7
2. io.spring.dependency-management to 1.1.0
3. nimbus-jose-jwt upgrade to version: '9.25'
4. Integration test cases
5. Functional Test cases
6. HELM CHART version: 4.0.11
7. APP_INSIGHTS_AGENT_VERSION=3.4.8

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by prudhvi-maddineni — APPROVED** [review]

**Review by AdilOoze — APPROVED** [review]

**Review by prudhvi-maddineni — DISMISSED** [review]

**Review by manukundloo-hmcts — APPROVED** [review]

File: lib/applicationinsights.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "connectionString": "${file:/mnt/secrets/rd/app-insights-connection-string}",
+  "role": {
+    "name": "RD Common Data API"
```
Comment (ishratt) [inline lib/applicationinsights.json:4]: is the name correct as this is caseworker api?

**Review by manukundloo-hmcts — DISMISSED** [review]

**Review by prudhvi-maddineni — APPROVED** [review]

**Review by AdilOoze — APPROVED** [review]

**Review by prudhvi-maddineni — DISMISSED** [review]

**Review by manukundloo-hmcts — DISMISSED** [review]

**Review by shivajihmcts — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -565,6 +562,12 @@ bootJar {
 }
 configurations.all {
   resolutionStrategy.eachDependency { details ->
+//    added netty-tcnative-boringssl-static because its required for
```
Comment (manukundloo-hmcts) [inline build.gradle:565]: Are we not deleting the commented line @vilasshelke-hmcts

**Review by manukundloo-hmcts — DISMISSED** [review]

**Review by prudhvi-maddineni — DISMISSED** [review]

**Review by prudhvi-maddineni — APPROVED** [review]

**Review by manukundloo-hmcts — DISMISSED** [review]

**Review by kavithagade12 — APPROVED** [review]

File: build.gradle [hunk 3]
```diff
@@ -379,14 +375,16 @@ dependencies {
...
+  implementation group: 'com.nimbusds', name: 'nimbus-jose-jwt', version: '9.25'
 
   implementation 'org.springframework.boot:spring-boot-starter-validation'
 
   implementation group: 'org.flywaydb', name: 'flyway-core', version: '6.5.5'
   implementation group: 'org.postgresql', name: 'postgresql', version: '42.5.1'
 
   implementation group: 'com.google.guava', name: 'guava', version: '31.1-jre'
+  //Added org.glassfish to support javax.el
+  implementation group: 'org.glassfish', name: 'javax.el', version: '3.0.0'
```
Comment (kavithagade12) [inline build.gradle:387]: Delete the commented piece of code

File: build.gradle [hunk 4]
```diff
@@ -408,10 +406,13 @@ dependencies {
...
-  implementation "com.github.hmcts.java-logging:logging-appinsights:${versions.reformLogging}"
+  implementation "com.github.hmcts.java-logging:logging-appinsights:5.1.9"
   implementation group: 'com.vladmihalcea', name: 'hibernate-types-52', version: '2.19.2'
   implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-jaxb-annotations', version: '2.13.4'
   implementation group: 'org.yaml', name: 'snakeyaml', version: '1.33'
+
+  implementation group: 'org.springdoc', name: 'springdoc-openapi-ui', version: '1.6.8'
+  implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-bootstrap', version: '3.1.5'
   // https://mvnrepository.com/artifact/net.minidev/json-smart
   implementation group: 'net.minidev', name: 'json-smart', version: '2.4.8'
```
Comment (kavithagade12) [inline build.gradle:417]: please remove commented mvn repo

**Review by kavithagade12 — APPROVED** [review]

**Review by manukundloo-hmcts — DISMISSED** [review]

**Review by manukundloo-hmcts — DISMISSED** [review]

**Review by ishratt — APPROVED** [review]

**Review by manukundloo-hmcts — APPROVED** [review]

**Pull request merged**